sdk/storage/azdatalake/file/client_test.go

```diff
@@ -3484,15 +3486,172 @@ func (s *RecordedTestSuite) TestFileAppendWithFlushOption() {
 
 	contentSize := 1024 * 8 // 8KB
 	rsc, _ := testcommon.GenerateData(contentSize)
+
+	opts := &file.AppendDataOptions{
+		LeaseAction:     &file.LeaseActionAcquire,
+		LeaseDuration:   to.Ptr(int64(15)),
+		ProposedLeaseID: proposedLeaseIDs[1],
+	}
+	_, err = srcFClient.AppendData(context.Background(), 0, rsc, opts)
+	_require.NoError(err)
+
+	gResp2, err := srcFClient.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(lease.StateTypeLeased, *gResp2.LeaseState)
+
+	time.Sleep(time.Second * 15)
+
+	//Check if the lease was acquired for the right duration
+	gResp, err := srcFClient.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(lease.StateTypeExpired, *gResp.LeaseState)
+}
+
+func (s *RecordedTestSuite) TestFileAppendDataWithRenewLease() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+
+	filesystemName := testcommon.GenerateFileSystemName(testName)
+	fsClient, err := testcommon.GetFileSystemClient(filesystemName, s.T(), testcommon.TestAccountDatalake, nil)
+	_require.NoError(err)
+	defer testcommon.DeleteFileSystem(context.Background(), _require, fsClient)
+
+	_, err = fsClient.Create(context.Background(), nil)
+	_require.NoError(err)
+
+	srcFileName := "src" + testcommon.GenerateFileName(testName)
+
+	srcFClient, err := testcommon.GetFileClient(filesystemName, srcFileName, s.T(), testcommon.TestAccountDatalake, nil)
+	_require.NoError(err)
+
+	createOpts := file.CreateOptions{
+		ProposedLeaseID: proposedLeaseIDs[0],
+		LeaseDuration:   to.Ptr(int64(15)),
+	}
+
+	resp, err := srcFClient.Create(context.Background(), &createOpts)
+	_require.NoError(err)
+	_require.NotNil(resp)
+
+	gResp2, err := srcFClient.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(lease.StateTypeLeased, *gResp2.LeaseState)
+
+	//Wait for 15 seconds for lease to expire
+	time.Sleep(15 * time.Second)
+
+	gResp, err := srcFClient.GetProperties(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(lease.StateTypeExpired, *gResp.LeaseState)
+
+	contentSize := 1024 * 8 // 8KB
+	rsc, _ := testcommon.GenerateData(contentSize)
+
+	opts := &file.AppendDataOptions{
+		LeaseAction:           &file.LeaseActionRenew,
+		LeaseAccessConditions: &file.LeaseAccessConditions{LeaseID: proposedLeaseIDs[0]},
+		LeaseDuration:         to.Ptr(int64(-1)),
```

Comment:
-1 means infinite lease, right? If yes, then does the cleanup happen when test completes.